src/Text/SlamSearch.purs

```diff
@@ -0,0 +1,36 @@
+module Text.SlamSearch where
+
+
+import Data.Semiring.Free
+import Data.String
+import Data.Foldable
+import Data.Either
+import Data.Array (reverse)
+import qualified Data.Semiring.Disjunctive as Dj
+
+import Text.SlamSearch.Types
+import qualified Text.SlamSearch.Parser.Tokens as Tk
+import qualified Text.SlamSearch.Parser as S
+import qualified Text.Parsing.Parser as P
+
+
+mkQuery :: String -> SearchQuery
+mkQuery input =
+  liftFree mkTerm $ prepare input
+  where prepare :: String -> Free String
+        prepare input = foldl (*) one $ free <$> reverse (split " " input)
+
+        
+        mkTerm :: String -> SearchQuery
+        mkTerm input =
+          either (const one) free $ parseTerm input
```

Comment:
Should this be `zero`? I think you'll end up with an empty term otherwise. Or even an error? It seems you're throwing parse errors away, or did I miss something?